### PR TITLE
AWS auto case - Validate Service Endpoints

### PR DESCRIPTION
**Pull request by yunjiang29**
Automate test case:
OCP-31378 - [ipi-on-aws][custom-region] Only one custom endpoint can be provided for each service.
OCP-31382	[ipi-on-aws][custom-region] Custom service endpoints should be HTTPS

New job: `aws-ipi-valid-endpoints-f14`
New workflow: `cucushift-installer-rehearse-aws-cases-valid-endpoints`

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-valid-endpoints-f14

**Comment by yunjiang29** [comment]
test get passed
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58168/rehearse-58168-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-valid-endpoints-f14/1849727465849622528/artifacts/aws-ipi-valid-endpoints-f14/cucushift-installer-rehearse-aws-cases-valid-endpoints-provision/build-log.txt

**Comment by yunjiang29** [comment]
@jianlinliu PTAL, thanks

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/valid-endpoints/provision/cucushift-installer-rehearse-aws-cases-valid-endpoints-provision-commands.sh [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    - name: ec2
+      url: http://ec2.${REGION}.amazonaws.com
+EOF
+yq-go m -x -i ${install_dir2}/install-config.yaml "${patch}"
+
+
+# ERROR failed to fetch Master Machines: failed to load asset "Install Config": failed to create install config: invalid "install-config.yaml" file: platform.aws.serviceEndpoints[0].url: Invalid value: "http://ec2.us-east-2.amazonaws.com": invalid scheme http, only https allowed
+openshift-install create manifests --dir ${install_dir2} || true
+expect_msg="invalid scheme http, only https allowed"
+if ! grep "${expect_msg}" ${install_dir2}/.openshift_install.log; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/valid-endpoints/provision/cucushift-installer-rehearse-aws-cases-valid-endpoints-provision-commands.sh:140]:
```suggestion
if ! grep -q "${expect_msg}" ${install_dir2}/.openshift_install.log; then
```

**Pull request merged**